src/components/MyJobs/WorkModeScreen.js

```diff
@@ -72,48 +74,106 @@ class WorkModeScreen extends Component {
 
   getJobHandler = (shift) => {
     LOG(`DEBUG:getJobHandler`, shift);
-    this.setState({ shift, isLoading: false });
+    this.setState({ shift, isLoading: false }, () => {
+      this.setDataTimes();
+    });
   };
 
   errorHandler = () => {
     this.setState({ isLoading: false });
   };
 
-  render() {
-    log(`DEBUG:state:`, this.state);
+  getJobRateHandler = (jobRate) => {
+    this.setState({ jobRate });
+  };
+
+  clockOutHandler = () => {
+    CustomToast(i18next.t('MY_JOBS.clockedOut'));
+    this.getClockins();
+  };
+
+  showRateButton = () => {
+    if (!this.state.shift) return false;
+
+    // check if current time is before ending_at
+    if (moment.utc().isSameOrBefore(moment.utc(this.state.shift.ending_at))) {
+      return false;
+    }
+
+    // check for missing clockouts or clockins
+    if (Array.isArray(this.state.clockIns)) {
+      if (!this.state.clockIns.length) return false;
+
+      const endedAt = this.state.clockIns[this.state.clockIns.length - 1]
+        .ended_at;
+
+      if (!endedAt) {
+        return false;
+      }
+    } else return false; // dont show rate button until clockins are loaded
+
+    if (Array.isArray(this.state.jobRate) && !this.state.jobRate.length) {
+      return true;
+    }
+
+    return false;
+  };
+
+  showAlreadyRated = () => {
+    if (!this.state.shift) return false;
+
+    if (Array.isArray(this.state.jobRate) && this.state.jobRate.length) {
+      return true;
+    }
+
+    return false;
+  };
+
+  setDataTimes = () => {
```

Comment:
@rich-97 cual es el proposito de sacar esto a un metodo? si quieres rebajar la complejidad de los calculos, usa una funcion en el archivo de job-actions.js Llenar el estado de valores calculados no es una buena practica, es mejor tener los valores con los cuales calcular, y calcular en el render. Ademas aqui estas introduciendo renders innecesarios

Reply:
Pero como hago para actualizar la barra cada segundo sin usar eso en el estado? o te refieres a sacar los calculos?

Reply:
`this.forceUpdate();` o `this.setState(this.state)`